### PR TITLE
[1.22][go1.16] Update to go1.16.7

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature
/area dependency release-eng release-eng/security security

#### What this PR does / why we need it:

Update Kubernetes to go1.16.7

Tracking issue: https://github.com/kubernetes/release/issues/2196

/assign @dims @liggitt @justaugustus 
cc: @kubernetes/release-engineering 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Kubernetes is now built with Golang 1.16.7
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cpanato** [comment]
/assign @dims @liggitt @justaugustus
cc: @kubernetes/release-engineering

File: build/dependencies.yaml [hunk 1]
```diff
@@ -124,7 +124,7 @@ dependencies:
       match: minimum_go_version=go([0-9]+\.[0-9]+)
 
   - name: "k8s.gcr.io/kube-cross: dependents"
-    version: v1.16.6-1
+    version: v1.16.7-1
```
Comment (dims) [inline build/dependencies.yaml:127]: `v1.16.7-0` or `v1.16.7-1`?

Comment (cpanato) [reply build/dependencies.yaml:127]: we built `v1.16.7-1` following the other, but I agree that should be `-0`
:/

Comment (cpanato) [reply build/dependencies.yaml:127]: https://github.com/kubernetes/release/pull/2197/files#diff-9d27d65f78671b29185b413d109ff0a70ddc2a5f7e5ca474ce6674224bff1ca1R19

Comment (justaugustus) [reply build/dependencies.yaml:127]: > `v1.16.7-0` or `v1.16.7-1`?

`v1.16.7-1` is correct. The `-1` is considered to be a "revision" number, similar to Debian revisions, starting at 1.

**Pull request merged**